pkg/cmd/roachtest/tests/failure_injection.go

```diff
@@ -0,0 +1,272 @@
+		blocked, err := checkPortBlocked(ctx, l, c, c.Nodes(1), c.Nodes(3))
+		if err != nil {
+			return err
+		}
+		if blocked {
+			return fmt.Errorf("expected connections from node 1 to node 3 to not be blocked")
+		}
+		return err
+	},
+}
+
+var asymmetricNetworkPartitionTest = failureSmokeTest{
+	testName:    "asymmetric network partition",
+	failureName: failures.IPTablesNetworkPartitionName,
+	args: failures.NetworkPartitionArgs{
+		Partitions: []failures.NetworkPartition{{
+			Source:      install.Nodes{1},
+			Destination: install.Nodes{3},
+			Type:        failures.Incoming,
+		}},
+	},
+	waitForFailureToPropagate: 5 * time.Second,
+	validateFailure: func(ctx context.Context, l *logger.Logger, c cluster.Cluster) error {
+		blocked, err := checkPortBlocked(ctx, l, c, c.Nodes(1), c.Nodes(3))
+		if err != nil {
+			return err
+		}
+		if blocked {
+			return fmt.Errorf("expected connections from node 1 to node 3 to be open")
+		}
+
+		blocked, err = checkPortBlocked(ctx, l, c, c.Nodes(3), c.Nodes(1))
+		if err != nil {
+			return err
+		}
+		if !blocked {
+			return fmt.Errorf("expected connections from node 3 to node 1 to be blocked")
+		}
+		return nil
+	},
+	validateRestore: func(ctx context.Context, l *logger.Logger, c cluster.Cluster) error {
+		blocked, err := checkPortBlocked(ctx, l, c, c.Nodes(1), c.Nodes(3))
+		if err != nil {
+			return err
+		}
+		if blocked {
+			return fmt.Errorf("expected connections from node 1 to node 3 to not be blocked")
+		}
+		return err
+	},
+}
+
+// Helper function that uses nmap to check if connections between two nodes is blocked.
+func checkPortBlocked(
+	ctx context.Context, l *logger.Logger, c cluster.Cluster, from, to option.NodeListOption,
+) (bool, error) {
+	// `nmap -oG` example output:
+	// Host: {IP} {HOST_NAME}	Status: Up
+	// Host: {IP} {HOST_NAME}	Ports: 26257/open/tcp//cockroach///
+	// We care about the port scan result and whether it is filtered or open.
+	res, err := c.RunWithDetailsSingleNode(ctx, l, option.WithNodes(from), fmt.Sprintf("nmap -p {pgport%[1]s} {ip%[1]s} -oG - | awk '/Ports:/{print $5}'", to))
+	if err != nil {
+		return false, err
+	}
+	return strings.Contains(res.Stdout, "filtered"), nil
+}
+
+func setupFailureSmokeTests(ctx context.Context, t test.Test, c cluster.Cluster) error {
+	// Download any dependencies needed.
+	if err := c.Install(ctx, t.L(), c.CRDBNodes(), "nmap"); err != nil {
+		return err
+	}
+	c.Start(ctx, t.L(), option.DefaultStartOpts(), install.MakeClusterSettings(), c.CRDBNodes())
+	// Run a light workload in the background so we have some traffic in the database.
+	c.Run(ctx, option.WithNodes(c.WorkloadNode()), "./cockroach workload init tpcc --warehouses=100 {pgurl:1}")
+	t.Go(func(goCtx context.Context, l *logger.Logger) error {
+		return c.RunE(goCtx, option.WithNodes(c.WorkloadNode()), "./cockroach workload run tpcc --tolerate-errors --warehouses=100 {pgurl:1-3}")
```

Comment:
With `--tolerate-errors`, it's essentially equivalent to `assert true`. Is that intended?

Reply:
Yeah my line of thinking here is that while it's possible our failure injection here does cause a legitimate cockroach issue, it's out of the scope of this test to catch it. i.e. Hopefully, any actual failures here will be caught on much more complex tests that do more than just spin up a TPCC workload. This way this test is more of a signal of "does the failure injection library work?" and less muddied by "does cockroach work under chaos?"

Reply:
Maybe `kv` workload without `--tolerate-errors` is a better option, then. Since `--tolerate-errors` is currently absolute in the sense that it ignores _all_ errors, in the _worst_ case, the workload may not even reach the network. Alternatively, we could just replace it with `--tolerate-retry-errors` [1]. (To my knowledge, the `kv` workload isn't susceptible to transaction retry errors.)

[1] https://github.com/cockroachdb/cockroach/issues/141792

Reply:
Switched to kv workload. Added benefit of not taking as long to set up so no complaints here 👍